src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -348,29 +412,89 @@ public sealed class FileInfoFormatting
             /// <summary>
             /// Gets or sets the style for directories.
             /// </summary>
-            public string Directory { get; set; } = "\x1b[44;1m";
+            public string Directory
+            {
+                get => _directory;
+                set => _directory = ValidateNoContent(value);
+            }
+
+            private string _directory = "\x1b[44;1m";
 
             /// <summary>
             /// Gets or sets the style for symbolic links.
             /// </summary>
-            public string SymbolicLink { get; set; } = "\x1b[36;1m";
+            public string SymbolicLink
+            {
+                get => _symbolicLink;
+                set => _symbolicLink = ValidateNoContent(value);
+            }
+
+            private string _symbolicLink = "\x1b[36;1m";
 
             /// <summary>
             /// Gets or sets the style for executables.
             /// </summary>
-            public string Executable { get; set; } = "\x1b[32;1m";
+            public string Executable
+            {
+                get => _executable;
+                set => _executable = ValidateNoContent(value);
+            }
+
+            private string _executable = "\x1b[32;1m";
+
+            /// <summary>
+            /// Custom dictionary handling validation of extension and content.
+            /// </summary>
+            public class FileExtensionDictionary : Dictionary<string, string>
+            {
+                /// <summary>
+                /// Initializes a new instance of the <see cref="FileExtensionDictionary"/> class.
+                /// </summary>
+                public FileExtensionDictionary() : base(StringComparer.OrdinalIgnoreCase) { }
+
+                /// <summary>
+                /// Add new extension and decoration to dictionary.
+                /// </summary>
+                /// <param name="extension">Extension to add.</param>
+                /// <param name="decoration">ANSI string value to add.</param>
+                public new void Add(string extension, string decoration)
+                {
+                    // if extension doesn't start with a dot, add one
+                    if (!extension.StartsWith("."))
```

Comment:
The actual formatting expects a period prefix due to simple check with Path.GetExtension.  Was thinking users might try: `$PSStyle.FileInfo.Extension.Add('zip',...)` without the period.  If the extension doesn't actually start with a period as stored in the dictionary, it'll never match up later in formatting.  Perhaps it's better to throw if the extension doesn't start with a period to avoid a `w.zip` type problem.

Reply:
Please use char overload extension.StartsWith('.').